### PR TITLE
RA-1855, update and delete relationships

### PR DESCRIPTION
**Pull request by cioan**
Hi @jwamalwa , here is my attempt on fixing this bug with the relationships controller. The code probably needs a bit more cleanup, but I was able to delete and update existing relationships on my local environment.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
@cioan  well done on this you could also add the ticket link on your pr description ,go on to claim the ticket too

**Comment by jwnasambu** [comment]
@HerbertYiga, The ticket link is Hyperlinked to RA-1855 and when you click or tap on it you will be directed to the issue.

**Comment by jwnasambu** [comment]
@cioan I have used your commit for testing since it has a single commits [RA-1717](https://github.com/openmrs/openmrs-module-registrationapp/pull/100),  and [1855](https://github.com/openmrs/openmrs-module-registrationapp/pull/101l) and everything works as expected.

**Review by sherrif10 — COMMENTED** [review]
LGTM cc @cioan @dkayiwa @mogoodrich

File: omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java [hunk 1]
```diff
@@ -227,39 +228,61 @@ private void addModelAttributes(PageModel model, Patient patient, Section sectio
...
                     }
                 }
             }
+
+            if (!CollectionUtils.isEmpty(existingRelationships)) {
+                //assume that any relationship that was not posted to this form it was deleted?
+                for (Relationship existingRelationship : existingRelationships) {
+                    personService.voidRelationship(existingRelationship, "deleted via the Registration widget");
+                }
+            }
```
Comment (mogoodrich) [inline omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java:284]: My apologies, as I'm stepping into something I haven't looked at in quite some time, but doesn't the relatioship widget allow you to customize the relationships you display to certain types?  If so, we need to make sure to apply the same logic when deleting (ie, don't delete types that you can't enter/view via the widget)

Comment (cioan) [reply omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java:284]: I don't think so, @mogoodrich , it shows all the patient relationships:
https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/webapp/resources/scripts/field/personRelationship.js#L12

This is an old person relationships widget that PIH is not using.

Comment (cioan) [reply omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java:284]: And here is the controller:
https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java#L223

Comment (mogoodrich) [reply omod/src/main/java/org/openmrs/module/registrationapp/page/controller/EditSectionPageController.java:284]: Sounds good to me, thanks @cioan !

**Review by mogoodrich — APPROVED** [review]
LGTM!

**Comment by cioan** [comment]
Thanks @mogoodrich !

**Pull request merged**